### PR TITLE
Remove the verb Toggle from the Block Inserter button.

### PR DESCRIPTION
**Pull request by afercia**
Addresses part of https://github.com/WordPress/gutenberg/issues/61483

## What?

- The verb 'toggle' isn't well translatable in many languages and should not be used.
- WordPress features should be title case, much like 'List View', 'Document Overview', and many other ones.

Note: I'd still like the 'Block Inserter' to be renamed because the current name is too technical and the tool isn't just inseerting blocks any longer. However, this is a first step to make the name better translatable and avoid the verb 'toggle'.

## Why?

- Better usability and localization.
- Consistency

## How?

- Chagees all the strings 'Toggle block inserter` to 'Block Inserter'.

## Testing Instructions

- Edit a post.
- Hover or focus the block inserter button. 
- Observe the tooltip is 'Block Inserter', title case.
- Check the same think in the Site Editor and in the Widgets editor.
- Make sure there are no occurrences of the string 'Toggle block inserter' aanywhere in the codebase.
- Check the tests pass.

### CODE REVIEW

**Review by draganescu — COMMENTED** [review]
Code wise it's a string replacement PR, let's wait for validation on the change itself. Otherwise great! I do agree with toggle being untranslateable but replacing it with just what the button is and not what it does makes me think maybe there is a better way? IDK tbh.

**Comment by kristastevens** [comment]
Hello! 👋 Editor here, answering the "needs copy review" ping. 

> Changes all the strings 'Toggle block inserter` to 'Block Inserter'.

^ No concerns with this proposed change.

**Comment by afercia** [comment]
Thanks for adding the missing issue labels. Looking into the failing tests.

I'd agree the name itself would needs some rethinking. This PR is just aiming to remove the verb 'toggle'.

**Comment by afercia** [comment]
I'm not sure I understand why the Performance test is failing `npm run test:performance -- post-editor.spec.js`
Any help would be appreciated Cc @WordPress/gutenberg-core

**Comment by SantosGuillamot** [comment]
I believe the problem wiht the failing tests is that there are two buttons matching the selector "Block Inserter": The actual "Block Inserter" and "Close Block Inserter".

To solve that, you can pass the property `exact: true`: [here](https://github.com/WordPress/gutenberg/pull/65983/files#diff-7ec757461fce833ba548841172950df439a392c3de98d773a64c77b2174ae429R372) & [here](https://github.com/WordPress/gutenberg/pull/65983/files#diff-7ec757461fce833ba548841172950df439a392c3de98d773a64c77b2174ae429R428) & [here](https://github.com/WordPress/gutenberg/pull/65983/files#diff-7ec757461fce833ba548841172950df439a392c3de98d773a64c77b2174ae429R487) & [here](https://github.com/WordPress/gutenberg/pull/65983/files#diff-7ec757461fce833ba548841172950df439a392c3de98d773a64c77b2174ae429R539). Something like this:

```js
const globalInserterToggle = page.getByRole( 'button', {
	name: 'Block Inserter',
	exact: true,
} );
```

**Comment by afercia** [comment]
Thanks @SantosGuillamot I did try `exact: true` already in https://github.com/WordPress/gutenberg/pull/65983/commits/32038377b2b484b6b7be10ad44f162610c5511eb and it didn't work. Maybe I missed a few places where to add it. Will try again.

**Comment by SantosGuillamot** [comment]
> To solve that, you can pass the property exact: true: [here](https://github.com/WordPress/gutenberg/pull/65983/files#diff-7ec757461fce833ba548841172950df439a392c3de98d773a64c77b2174ae429R372) & [here](https://github.com/WordPress/gutenberg/pull/65983/files#diff-7ec757461fce833ba548841172950df439a392c3de98d773a64c77b2174ae429R428) & [here](https://github.com/WordPress/gutenberg/pull/65983/files#diff-7ec757461fce833ba548841172950df439a392c3de98d773a64c77b2174ae429R487) & [here](https://github.com/WordPress/gutenberg/pull/65983/files#diff-7ec757461fce833ba548841172950df439a392c3de98d773a64c77b2174ae429R539).

I tested it locally adding `exact: true` in those links and that specific test passed.

**Comment by t-hamano** [comment]
I pushed a commit that attempts to solve the performance test.

In this PR, we expect the global inserter to have a label of "Block Inserter" in the `post-editor.spec.js` file.

But when the global inserter is open, there are two buttons that match this name. Screenshot:

inserter

So it seems like the error is happening here:

https://github.com/WordPress/gutenberg/blob/b691c1bc786d46e50241a34e322a1866d782e20b/test/performance/specs/post-editor.spec.js#L390

So I added a new `getByRole` to target only the button in the header.

I hope this approach solves the test, but I am concerned that this change might affect metrics at the same time.

**Comment by afercia** [comment]
When I run that sprecific test locally, the Gutenberg plugin gets ~disabled~ deactivated. As such, the editor uses the previous string `Toggle block inserter`. Anyone knows why the plugin gets ~disabled~ deactivated?

**Comment by afercia** [comment]
It appears the WordPress version used by the Gutenberg wp-env is 6.5 beta. This is what I see in the admin footer:
`You are using a development version (6.5-beta1-57638)..
It is not possible to activate latest Gutenberg plugin:
`Gutenberg requires WordPress 6.5 or later to function properly. Please upgrade WordPress before activating Gutenberg.`

Can anyone please shed a light on this? Cc @WordPress/gutenberg-core @desrosj 

Screenshot 2024-10-10 at 14 46 40

**Comment by t-hamano** [comment]
> I pushed a commit that attempts to solve the performance test.

It appears to have passed performance tests.

> So I added a new `getByRole` to target only the button in the header.
> I hope this approach solves the test, but I am concerned that this change might affect metrics at the same time.

If anyone knows if there are any concerns about this please let me know 🙏

**Comment by afercia** [comment]
> If anyone knows if there are any concerns about this please let me know 🙏

The tests pass but there are a few unanswered questions:
- Why `exact: true` didn't work in https://github.com/WordPress/gutenberg/commit/32038377b2b484b6b7be10ad44f162610c5511eb
- Why I'm still getting wp 6.5 beta locally when running the gutenberg wp-eng even after running `npm run wp-env start -- --update` I would have expected to get more recent images.

**Comment by t-hamano** [comment]
> Why `exact: true` didn't work in https://github.com/WordPress/gutenberg/commit/32038377b2b484b6b7be10ad44f162610c5511eb

From what I understand, the performance test runs `post-editor.spec.js` etc. on the trunk branch. In the trunk branch, the label of the global inserter is "Toggle block inserter", so adding `exact: true` will not match that label.

Furthermore, if there is no operation to explicitly close the main inserter in the e2e test, I think `exact: true` is unnecessary. Because when the main inserter is not open, the only button that matches the label "Block Inserter" is the main inserter toggle button.

> Why I'm still getting wp 6.5 beta locally when running the gutenberg wp-eng even after running `npm run wp-env start -- --update` I would have expected to get more recent images.

Have you specified a specific WordPress version via `.wp-env.override.json` or other means?

Normally, the latest version of WordPress should be enabled in both environments (`http://localhost:8888/`, `http://localhost:8889/`).

If there is no improvement, try `npm run wp-env destroy` and `npm run wp-env start`.

**Comment by afercia** [comment]
> Have you specified a specific WordPress version via .wp-env.override.json or other means?

Nope. 

> Normally, the latest version of WordPress should be enabled in both environments 

I'm not sure that is the case. As far as I can tell, the WordPress images of the Gutenberg wp-env need to be updated regularly.

**Comment by t-hamano** [comment]
> If there is no improvement, try `npm run wp-env destroy` and `npm run wp-env start`.

Have you tried that? Either way, if you're really having problems with wp-env, it's best to open a new issue.

**Comment by afercia** [comment]
@t-hamano thanks, I appreciate your feedback. I think I prefer to wait for an answer from the contributors I pinged on this PR. Thanks.

**Comment by afercia** [comment]
I'd appreciate a review to unblock this PR, thanks. Cc @WordPress/gutenberg-core

**Comment by talldan** [comment]
> So I added a new getByRole to target only the button in the header.
> I hope this approach solves the test, but I am concerned that this change might affect metrics at the same time.

Changes to the performance test code will need to be made at some points, so I expect it's ok. @youknowriad or @tyxla might know a bit more.

**Comment by ramonjd** [comment]
> Why I'm still getting wp 6.5 beta locally

When this happens to me I kill the images `npm run wp-env destroy`, and also remove the `.wp_env` directory on my local machine. Hope that helps!

For what it's worth, I tested this PR and things LGTM

| This PR | Trunk |
|--------|--------|
| Screenshot 2024-10-14 at 11 00 42 am | Screenshot 2024-10-14 at 11 00 37 am | 


I was wondering: when the block inspector is activated, is the button's `aria-expanded|pressed` state enough information to communicate that the button's action is now effectively "Close Block Inspector"?

**Review by tyxla — COMMENTED** [review]
> > So I added a new getByRole to target only the button in the header.
> > I hope this approach solves the test, but I am concerned that this change might affect metrics at the same time.
> 
> Changes to the performance test code will need to be made at some points, so I expect it's ok. @youknowriad or @tyxla might know a bit more.

While I'd generally say this should be fine because performance tests change every once in a whiile, it's definitely something to consider. Adding a `.getByRole()` could potentially increase the test by a few ms, which is a lot for a test that takes ~60ms on average. It might make sense to consider other ways to solve the ambiguity with a single `getByRole()`.   

One thing that comes to mind is using `exact: true` for that `getByRole()`. After all, one of the names is "Block Inserter" and the other is "Close Block Inserter", so with an exact match, you should be able to select only one of the buttons. I think that's what @SantosGuillamot proposed above, and I don't see why it wouldn't work.

**Comment by afercia** [comment]
After `npm run wp-env destroy` I tried again `exact: true` and all tests pass locally. Let's see i they pass on GitHub too.
I already tried `exact: true` in https://github.com/WordPress/gutenberg/pull/65983/commits/32038377b2b484b6b7be10ad44f162610c5511eb  and the tests failed for some reason.

**Comment by afercia** [comment]
> One thing that comes to mind is using `exact: true` for that `getByRole()`. After all, one of the names is "Block Inserter" and the other is "Close Block Inserter", so with an exact match, you should be able to select only one of the buttons. I think that's what @SantosGuillamot proposed above, and I don't see why it wouldn't work.

@tyxla [as you can see](https://github.com/WordPress/gutenberg/actions/runs/11325696407/job/31493126756?pr=65983), the tests fail on GitHub with `exact: true`. They pass locally. I'm assuming GitHub uses old images?

**Comment by tyxla** [comment]
> > One thing that comes to mind is using `exact: true` for that `getByRole()`. After all, one of the names is "Block Inserter" and the other is "Close Block Inserter", so with an exact match, you should be able to select only one of the buttons. I think that's what @SantosGuillamot proposed above, and I don't see why it wouldn't work.
> 
> @tyxla [as you can see](https://github.com/WordPress/gutenberg/actions/runs/11325696407/job/31493126756?pr=65983), the tests fail on GitHub with `exact: true`. They pass locally. I'm assuming GitHub uses old images?

Yes, I believe when the updated performance test suite runs against `trunk`, it fails because the rest of the changes from this PR aren't included. To get around this, I'd recommend making the performance test changes in a separate PR. 

Perhaps @youknowriad will have alternative ideas.

**Comment by afercia** [comment]
> To get around this, I'd recommend making the performance test changes in a separate PR.

@tyxla I'm not sure I understand how that would work. Removing the performance test changes from this PR wil make it fail. Am I missing something?

**Comment by tyxla** [comment]
@afercia sorry for the confusion. 

We might need to temporarily support both versions in the performance test spec.

That way the spec will work both with `trunk` and with this branch. 

See [this commit](https://github.com/WordPress/gutenberg/pull/62871/commits/73592062b62520058dff6f4527ab2197bc39ff02) for example of when this was done before.

Let me know if that makes sense.

**Comment by afercia** [comment]
> See [this commit](https://github.com/WordPress/gutenberg/pull/62871/commits/73592062b62520058dff6f4527ab2197bc39ff02) for example of when this was done before.

Thanks for clarifying @tyxla. So you suggest to use two locators and `.or()`. Given this discussion is all about impact on the performance test, would `.or()` impact more or less than the additional `getByRole` tried by @t-hamano?
Honestly I find this discussion a little derailing towards nitpicking, minor, aspects that are a little overkill for a very simple change like the one in this PR. Whatever your preference is please let me know and I'll implement it to unblock this PR, thanks.

**Comment by tyxla** [comment]
> > See [this commit](https://github.com/WordPress/gutenberg/pull/62871/commits/73592062b62520058dff6f4527ab2197bc39ff02) for example of when this was done before.
> 
> Thanks for clarifying @tyxla. So you suggest to use two locators and `.or()`. Given this discussion is all about impact on the performance test, would `.or()` impact more or less than the additional `getByRole` tried by @t-hamano? Honestly I find this discussion a little derailing towards nitpicking, minor, aspects that are a little overkill for a very simple change like the one in this PR. Whatever your preference is please let me know and I'll implement it to unblock this PR, thanks.

I definitely don't intend to block the PR. I was just trying to minimize the impact on performance tests which @t-hamano expressed concerns about. Feel free to go either way, I have no real preference, I was just trying to help.

**Comment by afercia** [comment]
Thanks. I reverted the last commit. The approach with two locators and `.or()` seems a little too verbose to me and it would need to be implemented in 4 tests. @t-hamano approach looks way simpler to me. Waiting for the tests to pass.

**Comment by youknowriad** [comment]
What I can say about the performance tests is that they are supposed to run on multiple branches at the same time. So yeah, we need to make sure selectors we use can work both for 6.7 and 6.8 (trunk and this PR).

Also, from the changes that I see there, It doesn't seem that these changes have any impact on the metrics themselves, they are part of the "preparation" steps of the tests so they won't impact the numbers AFAIK.

**Comment by afercia** [comment]
Thanks @youknowriad for clarifying an important point.
A final review and approval would be appreciated, when the performance test completes and passes 🙏🏻

**Review by t-hamano — APPROVED** [review]
@youknowriad Thanks for the feedback!

If this PR doesn't impact the metrics, I believe the current approach is best.

**Pull request merged**
